pkg/storage/replica.go

```diff
@@ -1003,44 +1003,46 @@ func (r *Replica) redirectOnOrAcquireLease(ctx context.Context) (LeaseStatus, *r
 		}
 
 		// Wait for the range lease to finish, or the context to expire.
-		slowTimer := time.After(base.SlowRequestThreshold)
-		select {
-		case pErr = <-llChan:
-			if pErr != nil {
-				// Getting a LeaseRejectedError back means someone else got there
-				// first, or the lease request was somehow invalid due to a
-				// concurrent change. Convert the error to a NotLeaseHolderError.
-				if _, ok := pErr.GetDetail().(*roachpb.LeaseRejectedError); ok {
-					lease, _ := r.getLease()
-					var err error
-					if !r.IsLeaseValid(lease, r.store.Clock().Now()) {
-						err = newNotLeaseHolderError(nil, r.store.StoreID(), r.Desc())
-					} else {
-						err = newNotLeaseHolderError(lease, r.store.StoreID(), r.Desc())
+		pErr = func() *roachpb.Error {
```

Comment:
This is just code movement. The big difference is that it's wrapped in a method and there's a for loop in order to continue waiting after handling the slow timer signaling.